docs/proposals/modelcaching.md

```diff
@@ -0,0 +1,102 @@
+# Model Caching Proposal
+
+## Summary
+
+This proposal introduces a new llm-d component controlling the location of models to make vLLM startup time faster and more reliable. It also includes an implementation design focusing on caching models to local storage, with plans to expand functionalities based on community feedback.
+
+# Motivation
+
+vLLM startup time can greatly be reduced when models are cached on local storage. All supported vLLM model loaders, safetensor (builtin), [fastsafetensors](https://docs.vllm.ai/en/latest/models/extensions/fastsafetensor.html?h=fastsafete), [Run.ai model streamer](https://docs.vllm.ai/en/latest/models/extensions/runai_model_streamer.html), [Coreweave’s tensoriser](https://docs.vllm.ai/en/latest/models/extensions/tensorizer.html) support loading models from local storage. Each has consistently shown faster startup compared to loading models from HuggingFace (see [\#1](https://github.com/run-ai/runai-model-streamer/blob/master/docs/src/benchmarks.md), [\#2](https://github.com/vllm-project/vllm/pull/10647), [\#3](https://v1.docs.coreweave.com/coreweave-machine-learning-and-ai/inference/tensorizer#benchmarks)).
+
+In addition, loading models from local storage is considered more reliable than loading from the network (eg. S3, Huggingface, etc...), as it reduces dependency on external services, mitigates issues related to connectivity and latency, and makes system operations more predictable.
+
+# Goals
+
+* Be agnostic about which model loader is being used.
+* Provide a low-level API to cache models on local storage.
+
+# Non-Goals
+
+* Provisioning Persistent Volumes, which is better handled by existing external static provisioners, such as [Local Persistent Volume Static Provisioner](https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner), [Openshift Local Storage Operator](https://catalog.redhat.com/software/container-stacks/detail/66993e0f4f284e980ee072d3) and [Open EBS](https://openebs.io).
+* Supporting other storage types such as S3, HTTP, etc…
+* Supporting model revision (GIE does not support it yet).
+* Providing an high-level CRD-based API.
```

Comment:
It seems to me that defined structured data would be preferable to annotations on PVCs.